techniques/systemSettings/userManagement/userManagement/10/userManagement.ps1.st

```diff
@@ -91,23 +96,36 @@ function check_usergroup_user_parameters_&RudderUniqueID& {
         $methodCallPresent = User-Absent -login $logins[$i] -PolicyMode $policyMode
         $null = Compute-Method-Call -MethodCall $methodCallPresent @userParams
 
-#To REMOVE in the future
-      } elseif ($policies[$i] -eq $check_present) {
-        $methodCallPresent = User-Present -login $logins[$i] -PolicyMode ([Rudder.PolicyMode]::Audit)
-        $null = Compute-Method-Call -MethodCall $methodCallPresent @userParams
-      } elseif ($policies[$i] -eq $check_absent) {
-        $methodCallPresent = User-Absent -login $logins[$i] -PolicyMode ([Rudder.PolicyMode]::Audit)
-        $null = Compute-Method-Call -MethodCall $methodCallPresent @userParams
+      } elseif ( (-not $userExists) -and ($policies[$i] -eq $updateIfExists) ) {
+        $null = Rudder-Report-NA  @userParams -message "User does not exists"
+        $null = Rudder-Report-NA  @passwordParams -message "User does not exists"
       }
 
       #Unix section reporting
-      $unixParams = $commonParams + @{
-          ComponentName = "Home directory"
+      $primaryGroup = $commonParams + @{
+          ComponentName = "User primary group"
+          ComponentKey = $logins[$i]
+      }
```

Comment:
You can create a `naParams` variable based on the `commonParams` variable and extend it with the message and componentKey. Something like:
``
$naParams = $commonParams  + @{
  ComponentKey = $logins[$i]
  Message = 'Not applicable on Windows'
}
$null = Rudder-Report-NA @naParams -ComponentName 'User primary group'
$null = Rudder-Report-NA @naParams -ComponentName 'User full name'
$null = Rudder-Report-NA @naParams -ComponentName 'User default shell'
$null = Rudder-Report-NA @naParams -ComponentName 'Home Directory'